### PR TITLE
Routing and Movie search

### PR DESCRIPTION
**Pull request by melk0sha**
**Task 8. Routing and Movie search**

Added 404 and “No movies found” pages with markup. Have 2+ pages which displays on different URLs.
Implemented displaying 404 page for all invalid URLs.
By default, user lands on a new page with empty results page.
When user performs a search on the page, changed URL and showed search results. When new user enters the site using direct link with search parameters – showed search results.

### CODE REVIEW

File: src/components/findMovieSection/findMovieSection.jsx [hunk 1]
```diff
@@ -42,4 +78,20 @@ const FindMovieSection = () => {
   );
 };
 
-export default FindMovieSection;
+FindMovieSection.propTypes = {
+  moviesSortBy: moviesSortByType,
+  onMoviesByParamsGet: func,
+  searchValue: string
+};
```
Comment (cate-blint) [inline src/components/findMovieSection/findMovieSection.jsx]: onSearchValueChange  type is missed

File: src/components/movieSection/movieSection.jsx [hunk 2]
```diff
@@ -30,13 +30,13 @@ const MovieSection = ({ movie = {} }) => {
...
+              {!!movie.vote_average && <MovieRating>{movie.vote_average}</MovieRating>}
             </InfoWrapper>
-            <MovieSpan>{movie.tagline}</MovieSpan>
+            {!!movie.tagline && <MovieSpan>{movie.tagline}</MovieSpan>}
             <MovieGenres>{genres}</MovieGenres>
             <InfoWrapper>
-              <MovieSpan>{year}</MovieSpan>
-              <MovieSpan>{movie.runtime} min</MovieSpan>
+              {!!year && <MovieSpan>{year}</MovieSpan>}
+              {!!movie.runtime && <MovieSpan>{movie.runtime} min</MovieSpan>}
```
Comment (cate-blint) [inline src/components/movieSection/movieSection.jsx:56]: Why did you add these checks here?

Comment (melk0sha) [reply src/components/movieSection/movieSection.jsx:56]: sometimes we have no value and can see something like " min" without any value or "0" rating value.

File: src/components/findMovieSection/findMovieSection.jsx (hunk 1 above)
Comment (melk0sha) [reply src/components/findMovieSection/findMovieSection.jsx]: fixed
